src/components/Navigation/index.js

```diff
@@ -60,152 +54,139 @@ export default class Navigation extends Component {
-    if(window.screen.width < mediumContainerWidth) {
-      this.setState({
-        isOpen: false,
-      })
-    } else {
-      this.setState({
-        isOpen: true,
-      })
-    }
-  }
-
   render() {
 
-    const {instructor, items} = this.props
+    const {instructor, items, isLikelyDesktop} = this.props
     const {isOpen} = this.state
 
     return (
-      <ContainerWidth onWidthChange={this.handleWidthChange}>
-        {(screenSize) => (
-          <aside
-            className='bg-base fixed min-vh-100 h-100 z-1 pt2-s'
-            style={{
-              width: navigationWidth,
-              willChange: 'transform',
-              transition: 'transform .3s',
-              left: 0,
-              transform: isOpen ? 'translateX(0)' : 'translateX(-100%)',
-            }}
-          >
-
-            <div
-              onClick={this.toggle.bind(this)}
-              className={`dn-ns fixed pa2 br2 br--right br--bottom bg-base ${isOpen ? '' : 'o-30'}`}
-              style={{
-                marginLeft: navigationWidth,
-              }}
-            >
-              <Icon
-                type={isOpen ? 'close' : 'menu'}
-                size='3'
+      <aside
+        className='bg-base fixed min-vh-100 h-100 z-1 pt2-s'
+        style={{
+          width: navigationWidth,
+          willChange: 'transform',
+          transition: 'transform .3s',
+          left: 0,
+          transform: isOpen ? 'translateX(0)' : 'translateX(-100%)',
+        }}
+      >
+
+        <div
+          onClick={this.toggle.bind(this)}
+          className={`fixed pa2 br2 br--right br--bottom bg-base ${isOpen ? '' : 'o-30'}`}
+          style={{
+            marginLeft: navigationWidth,
+            display: isLikelyDesktop
+              ? 'none'
+              : 'block'
+          }}
+        >
+          <Icon
+            type={isOpen ? 'close' : 'menu'}
+            size='3'
+            color='white'
+          />
+        </div>
+
+        <div className='pa2 pa3-ns bb tc b--dark-blue'>
+          <EggoInstructorBanner />
+        </div>
+
+        <div className='tc ph3 pt4 pb3 dn db-ns'>
+          <Avatar
+            name={instructor.full_name}
+            url={instructor.avatar_url}
+          />
+          <div className='mt2 white'>
+            {instructor.first_name}
+          </div>
+        </div>
+
+        <nav className={`
+          pv2
+          ${isOpen
+            ? 'flex flex-column'
+            : 'dn'
+          }
+          flex flex-column
+        `}>
+          {map(items, (item, index) => {
+
+            const NavigationLinkContents = () => (
+              <IconLabel
+                iconType={item.iconType}
+                labelText={item.text}
                 color='white'
               />
-            </div>
-
-            <div className='pa2 pa3-ns bb tc b--dark-blue'>
-              <EggoInstructorBanner />
-            </div>
+            )
+
+            if(isFunction(item.action)) {
```

Comment:
@ijones16 Thanks for letting me know; this logic lets you call any "action" from a navigation item click (so run a function, link to another website, or link to a relative link with react router); see here:
https://github.com/eggheadio/egghead-instructor-center/blob/master/src/index.js#L69-L103

![image](https://cloud.githubusercontent.com/assets/5497885/25596848/9cb39d40-2e88-11e7-9b96-7bbd54b876be.png)

So, `Log Out` calls a function, `Guide` links to another website (the instructor guide), `Lessons` links to the relative `/lessons` link inside the app.

Make sense?